src/coursescheduler/models.py

```diff
@@ -1,59 +1,110 @@
+    "tuesday": [TimeRange],
+    "wednesday": [TimeRange],
+    "thursday": [TimeRange],
+    "friday": [TimeRange],
+})
+
+Professor = Schema({
+    "id": str,
+    "name": str,
+    "isPeng": bool,
+    "facultyType": Or("RESEARCH", "TEACHING"),
+    "coursePreferences": [CoursePreference],
+    "teachingObligations": int,
+    "preferredTimes": {
+        "fall": Or(None, DayTimes),
+        "spring": Or(None, DayTimes),
+        "summer": Or(None, DayTimes),
     },
-    "course code2": {
-        "pengRequired": "boolean",
-        "yearRequired": "int",
-        "semester": "string",
-        "professor": "professor object",
-        "timeSlots": "list of timeSlot objects",
-        "academicYear": "int"
+    "preferredCoursesPerSemester": {
+        "fall": And(int, lambda n: n >= 0),
+        "spring": And(int, lambda n: n >= 0),
+        "summer": And(int, lambda n: n >= 0),
     },
-}
-
-# A dictionary of professors where the professor ids are the keys and the values are the attributes of each professor
-professors = {
-    "id": {
-        "name": "string",
-        "isPeng": "boolean",
-        "facultyType": "string",
-        "coursePreferences": "list of coursePreference objects",
-        "teachingObligations": "int",
-        "preferredTimes": "dictionary of DayTimes objects",
-        "preferredCoursesPerSemester": "dictionary of fall, spring, and summer ints",
-        "preferredNonTeachingSemester": "string",
-        "preferredCourseDaySpreads": "a course day spread object"
+    "preferredNonTeachingSemester": Or("FALL", "SPRING", "SUMMER", None),
+    "preferredCourseDaySpreads": ["TWF", "MTh", "M", "T", "W", "Th", "F"],
+})
+
+Course = Schema({
+    "code": str,
+    "title": str,
+    "pengRequired": {
+        "fall": bool,
+        "spring": bool,
+        "summer": bool,
     },
-}
-
-# Basic example of the time slot configuration.
-# Below is a possible solution change if the time slot dictionary is too slow
-# Conflict determination: IF list is same length, only need to compare 1 value
-# First check if the day overlaps, then check if time overlaps
-# MR and TWF will never need to be compared
-timeslots = {
-    1: [["TUESDAY", "830", "920"], ["WEDNESDAY", "830", "920"], ["FRIDAY", "830", "920"]],
-    2: [["TUESDAY", "900", "950"], ["WEDNESDAY", "900", "950"], ["FRIDAY", "900", "950"]],
-    3: [["MONDAY", "900", "1020"], ["THURSDAY", "900", "1020"]],
-    4: [["TUESDAY", "900", "1150"]]
-}
-
-
-# conflicts = {
-#     1: ABC,
-#     2: BC
-# }
-# set_1.isdisjoint(set_2)
-# domain = [1, 2, 3, 4, 5, . . . . . ]
-
-# time_conflicts = {(id1, id2) : bool does_conflict}
+    "yearRequired": Or(1, 2, 3, 4),
+})
+
+TimeSlot = Schema({
+    "dayOfWeek": Or("MONDAY", "TUESDAY", "WEDNESDAY", "THURSDAY", "FRIDAY"),
+    "timeRange": TimeRange,
+})
+
+ProfessorSlim = Schema({
+    "id": str,
+    "name": str,
+})
+
+CourseSection = Schema({
+    "professor": Or(None, ProfessorSlim),
+    Optional("maxCapacity"): Or(None, int),
```

Comment:
If maxCapacity is being added to our spec, should it be marked optional here?

Reply:
It should be optional; it is not going to be added to the spec